src/tasks/get-own-plan-task.ts

```diff
@@ -0,0 +1,44 @@
+import { BaseTask } from "./base-task";
+import { Stripe } from "stripe";
+import { DatabaseTransactionHandler, Member } from "graasp";
+import { FastifyLoggerInstance } from "fastify";
+import { Plan } from "../interfaces/plan";
+import { CustomerExtra } from "../interfaces/customer-extra";
+
+export class GetOwnPlanTask extends BaseTask<Plan> {
+  get name(): string {
+    return GetOwnPlanTask.name;
+  }
+
+  constructor(member: Member<CustomerExtra>, stripe: Stripe) {
+    super(member, null, stripe);
+  }
+
+  async run(handler: DatabaseTransactionHandler, log: FastifyLoggerInstance): Promise<void> {
+    this.status = "RUNNING";
+
+    const {
+      extra: { subscriptionId },
+    } = this.actor;
+
+    const subscription = await this.stripe.subscriptions.retrieve(subscriptionId, {
+      expand: ["items.data.price.product"],
+    });
+
+    const price = subscription.items.data[0].price;
+
+    const plan = {
+      id: price.id,
+      name: (<Stripe.Product>price.product).name,
+      price: price.unit_amount / 100 ?? 0,
```

Comment:
OK. Here the comment I left above becomes clearer. I assume that Stripe is using integers for pricing. Why are you dividing here? Backend should keep prices in integers for accuracy. We can safely assume we will only charge up to two decimals, for simplicity's sake.

Reply:
The frontend then takes care of displaying the decimal.